handlers/package.js

```
@@ -15,6 +15,24 @@ package.show = function(request, reply) {
   request.logger.info('get package: ' + name);
 
   var promise = Package.get(name)
+    .catch(function(err){
+      if (err.statusCode === 404) {
+        if (validatePackageName(name).validForNewPackages) {
+          context.package = {name: name}
+          request.logger.error('package not found: ' + name);
+          reply.view('errors/not-found', context).code(404);
+          return promise.cancel();
+        }
+
+        request.logger.error('invalid package name: ' + name);
+        reply.view('errors/not-found', context).code(400);
+        return promise.cancel();
+      }
+
+      request.logger.error(err);
+      reply.view('errors/internal', context).code(500);
+      return promise.cancel();
```

In some cases you want to ignore the error and trickle up to the next `then`, but in other cases you want to abort all the things?